http.go

```diff
@@ -50,15 +50,15 @@ func NewHttpClient(ctx context.Context, cfg HasClientConfig, vault HasVaultConfi
 		cli: &http.Client{
 			Transport: &http.Transport{
 				TLSClientConfig: ct,
-				IdleConnTimeout: cfg.GetConnTimeout(),
-				WriteBufferSize: cfg.GetWriteBufferSize(),
-				ReadBufferSize:  cfg.GetReadBufferSize(),
+				IdleConnTimeout: cfg.ConnTimeout,
+				WriteBufferSize: cfg.WriteBufferSize,
+				ReadBufferSize:  cfg.ReadBufferSize,
 			},
 		},
 	}
 
-	if cfg.GetToken() != nil && cfg.GetToken().GetShared() != nil {
-		if httpCli.token, err = LoadKey(ctx, cfg.GetToken().GetShared(), vault, "shared"); err != nil {
+	if cfg.Token.Shared.GetEnabled() {
```

Comment:
Dont we need a null check for cfg.Token?

Reply:
No longer possible to be nil.